src/package-manager.ts

```diff
@@ -99,13 +102,16 @@ export async function detectPackageManager(
               packageManagers.find(
                 (pm) => pm.name === name && pm.majorVersion === majorVersion,
               ) || packageManagers.find((pm) => pm.name === name);
-            return {
-              ...packageManager,
-              name,
```

Comment:
This is by design. we return the limited information we can infer from all possible sources. `packageManager` could be `xyz@1.2.3` (where `xyz` is not defined in nypm, yet), we return limited info about it.